modules/database/src/std/link/lnkConst.c

```diff
@@ -225,8 +225,10 @@ static jlif_result lnkConst_string(jlink *pjlink, const char *val, size_t len)
         if (!vec)
             return jlif_stop;
         str = malloc(len+1);
-        if (!str)
+        if (!str) {
+            free(vec);
```

Comment:
Agreed.

Reply:
Looking at this logic, I don't think it was or is handling OOM correctly.  By the time of the `malloc()`, the prceeding `realloc()` may have invalidated `clink->value.pmem`.  The added `free(vec)` would now make certain that `clink->value.pmem` is invalid, and left as a dangling pointer.

A better scheme might be to do the `malloc()` first.  like:

```c
str = malloc(len+1);
if(!str)
    return jlif_stop;
vec = realloc(clink->value.pmem, newElems * sizeof(char *));
if(!vec) {
    free(str);
    return jlif_stop; /* clink->value.pmem remains valid */
}
```

Reply:
How can `realloc()` invalidate that buffer in an OOM situation? If it can't supply the additional memory requested it [must](https://en.cppreference.com/w/c/memory/realloc) return `NULL` without freeing or modifying the old block. The `NULL` will then stop the parsing (line 225).

By this point `clink->type == ac40` so `lnkConst_free()` will eventually `free()` each of the already-saved strings and the `clink->value.pmem` buffer. The `clink->nElems` value is not incremented until later when `lnkConst_string()` is about to `return jlif_continue;` so `lnkConst_free()` won't try to free the uninitialized value that has been appended to the vector.

I agree that the original change that @JJL772 proposed is not correct either, the solution should be to move the assignment of `clink->value.pstrings = vec;` up to just before the call to `malloc()`, which will ensure that the new vector and all of the previously assigned string elements in it get properly released in `lnkConst_free()`.